plugins/node/instrumentation-react-native-navigation/test/NativeNavigationTracker.test.tsx

```diff
@@ -0,0 +1,223 @@
+import * as rnn from './helpers/react-native-navigation';
+
+const AppWithProvider = ({ shouldPassProvider = true }) => {
+  const { Navigation } = rnn;
+  const provider = useProvider();
+  const ref = useRef(Navigation.events());
+
+  return (
+    <NativeNavigationTracker
+      ref={ref}
+      provider={shouldPassProvider ? provider.current : undefined}
+    >
+      my app goes here
+    </NativeNavigationTracker>
+  );
+};
+
+describe('NativeNavigationTracker.tsx', function () {
+  const sandbox = sinon.createSandbox();
+  const mockDidAppearListener = sandbox.spy();
+  const mockDidDisappearListener = sandbox.spy();
+
+  let mockConsoleDir: sinon.SinonSpy;
+  let mockAddEventListener: sinon.SinonSpy;
+
+  beforeEach(function () {
+    const { Navigation } = rnn;
+    sandbox.stub(Navigation, 'events').returns({
+      registerComponentDidAppearListener: mockDidAppearListener,
+      registerComponentDidDisappearListener: mockDidDisappearListener,
+    });
+
+    mockAddEventListener = sandbox.spy(AppState, 'addEventListener');
+    mockConsoleDir = sandbox.spy(console, 'dir');
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  it('should render a component that implements <NativeNavigationTracker /> without passing a provider', async function () {
+    // const screen =
+    render(<AppWithProvider shouldPassProvider={false} />);
+
+    sandbox.assert.calledWith(mockDidAppearListener, sandbox.match.func);
+
+    const mockDidAppearListenerCall = mockDidAppearListener.getCall(0).args[0];
+
+    mockDidAppearListenerCall({ componentName: 'initial-test-view' });
+    sandbox.assert.calledWith(mockDidDisappearListener, sandbox.match.func);
+
+    const mockDidDisappearListenerCall =
+      mockDidDisappearListener.getCall(0).args[0];
+
+    mockDidDisappearListenerCall({ componentName: 'initial-test-view' });
+
+    sandbox.assert.calledWith(
+      mockConsoleDir,
+      sandbox.match({
+        name: 'initial-test-view',
+        traceId: sandbox.match.string,
+        attributes: {
+          [ATTRIBUTES.initialView]: true,
+          [ATTRIBUTES.appState]: 'active',
+        },
+        timestamp: sandbox.match.number,
+        duration: sandbox.match.number,
+      }),
+      sandbox.match({ depth: sandbox.match.number })
+    );
+
+    mockDidAppearListenerCall({ componentName: 'second-test-view' });
+    mockDidDisappearListenerCall({ componentName: 'second-test-view' });
+
+    sandbox.assert.calledWith(
+      mockConsoleDir,
+      sandbox.match({
+        name: 'second-test-view',
+        traceId: sandbox.match.string,
+        attributes: {
+          [ATTRIBUTES.initialView]: false,
+          [ATTRIBUTES.appState]: 'active',
+        },
+        timestamp: sandbox.match.number,
+        duration: sandbox.match.number,
+      }),
+      sandbox.match({ depth: sandbox.match.number })
+    );
+
+    // sandbox.assert.match(screen.getByText('my app goes here'), true);
```

Comment:
I would like to reintroduce this assert to make sure whatever we wrap into our component is mounted without issues. also it would be great to add comments explaining the test itself (both apply to the 2nd test file)